src/include/planner/logical_plan/logical_operator/logical_semi_masker.h

```diff
@@ -9,24 +9,31 @@ namespace planner {
 
 class LogicalSemiMasker : public LogicalOperator {
 public:
-    LogicalSemiMasker(LogicalScanNode* scanNode, std::shared_ptr<LogicalOperator> child)
-        : LogicalOperator{LogicalOperatorType::SEMI_MASKER, std::move(child)}, scanNode{scanNode} {}
+    LogicalSemiMasker(std::shared_ptr<binder::Expression> nodeID,
+        std::vector<LogicalOperator*> scanNodes, std::shared_ptr<LogicalOperator> child)
+        : LogicalOperator{LogicalOperatorType::SEMI_MASKER, std::move(child)},
+          nodeID{std::move(nodeID)}, scanNodes{std::move(scanNodes)} {}
 
     inline void computeFactorizedSchema() override { copyChildSchema(0); }
     inline void computeFlatSchema() override { copyChildSchema(0); }
 
     inline std::string getExpressionsForPrinting() const override {
-        return scanNode->getNode()->toString();
+        return ((LogicalScanNode*)scanNodes[0])->getNode()->toString();
     }
 
-    inline LogicalScanNode* getScanNode() const { return scanNode; }
+    inline std::shared_ptr<binder::Expression> getNodeID() const { return nodeID; }
+    inline bool isMultiLabel() const {
+        return ((LogicalScanNode*)scanNodes[0])->getNode()->isMultiLabeled();
```

Comment:
Ditto, here why not directly use `nodeID`?

Reply:
Because nodeID is a propertyExpression and doesn't have isMultiLabeled() interface.